lib/iris/experimental/representation.py

```diff
@@ -0,0 +1,308 @@
+        self.cube_id = id(self.cube)
+        self.cube_str = str(self.cube)
+
+        self.str_headings = {
+            'Dimension coordinates:': None,
+            'Auxiliary coordinates:': None,
+            'Derived coordinates:': None,
+            'Scalar coordinates:': None,
+            'Attributes:': None,
+            'Cell methods:': None,
+        }
+        self.dim_desc_coords = ['Dimension coordinates:',
+                                'Auxiliary coordinates:',
+                                'Derived coordinates:']
+
+        # Important content that summarises a cube is defined here.
+        self.shapes = self.cube.shape
+        self.scalar_cube = self.shapes == ()
+        self.ndims = self.cube.ndim
+
+        self.name = self.cube.name().title().replace('_', ' ')
+        self.names = self._dim_names()
+        self.units = self.cube.units
+
+    def _get_dim_names(self):
+        """
+        Get dimension-describing coordinate names, or '--' if no coordinate]
+        describes the dimension.
+
+        Note: borrows from `cube.summary`.
+
+        """
+        # Create a set to contain the axis names for each data dimension.
+        dim_names = list(range(len(self.cube.shape)))
+
+        # Add the dim_coord names that participate in the associated data
+        # dimensions.
+        for dim in range(len(self.cube.shape)):
+            dim_coords = self.cube.coords(contains_dimension=dim,
+                                          dim_coords=True)
+            if dim_coords:
+                dim_names[dim] = dim_coords[0].name()
+            else:
+                dim_names[dim] = '--'
+        return dim_names
+
+    def _dim_names(self):
+        if self.scalar_cube:
+            dim_names = ['(scalar cube)']
+        else:
+            dim_names = self._get_dim_names()
+        return dim_names
+
+    def _get_lines(self):
+        return self.cube_str.split('\n')
+
+    def _get_bits(self, bits):
+        """
+        Parse the body content (`bits`) of the cube string in preparation for
+        being converted into table rows.
+
+        """
+        left_indent = re.split(r'\w+', bits[1])[0]
+
+        # Get heading indices within the printout.
+        start_inds = []
+        for hdg in self.str_headings.keys():
+            heading = '{}{}'.format(left_indent, hdg)
+            try:
+                start_ind = bits.index(heading)
+            except ValueError:
+                continue
+            else:
+                start_inds.append(start_ind)
+        # Mark the end of the file.
+        start_inds.append(0)
+
+        # Retrieve info for each heading from the printout.
+        for i0, i1 in zip(start_inds[:-1], start_inds[1:]):
+            str_heading_name = bits[i0].strip()
+            if i1 != 0:
+                content = bits[i0 + 1: i1]
+            else:
+                content = bits[i0 + 1:]
+            self.str_headings[str_heading_name] = content
+
+    def _make_header(self):
```

Comment:
Would be great to have column highlighting too. See https://stackoverflow.com/questions/16967366/how-do-i-highlight-an-html-table-column-with-bootstrap.

Note: not in this PR.